### PR TITLE
Night mode

### PR DESCRIPTION
**Pull request by Fodilahy-mena**
Hello team!
I have added some functionalities to the night mode tool button for switching between dark mode and default mode with entire app theme colors and background. Please, have a look at it when you are available and leave comments if necessary.

- Branch link: https://github.com/onja-org/malagasy-rn-Polonium/tree/night-mode
- Click up link: https://app.clickup.com/t/35bjc2
- Refs: U11 #35bjc2

Thanks in advance

### CODE REVIEW

**Comment by rinonTen** [comment]
You'll not be able to merge this branch after our approvals because you still have conflicts to resolve. 
Good work though!

File: src/components/Textarea/Textarea.js
Comment (rinonTen) [inline src/components/Textarea/Textarea.js]: It would be more readable and efficient if you have variables for the colors because you're repeating the same color a few times.

File: src/components/Textarea/Textarea.js
Comment (rinonTen) [inline src/components/Textarea/Textarea.js]: This one

File: src/components/Textarea/Textarea.js
Comment (rinonTen) [inline src/components/Textarea/Textarea.js]: The same here

File: src/redux/theme/index.js
Comment (rinonTen) [inline src/redux/theme/index.js:22]: This is really good but I would use variables for those colors. It's not necessary though

File: src/containers/HomeScreenContainer.js
Comment (andrzej-onja) [inline src/containers/HomeScreenContainer.js:31]: newTerms ?

File: src/redux/actions/index.js
Comment (andrzej-onja) [inline src/redux/actions/index.js]: remove and just use set

File: src/redux/reducers/index.js
Comment (andrzej-onja) [inline src/redux/reducers/index.js]: remove get

File: src/redux/actions/index.js
Comment (andrzej-onja) [inline src/redux/actions/index.js]: export function switchTheme() {
  return {
    type: SWITCH_THEME,
  };
}

File: src/redux/reducers/index.js
Comment (andrzej-onja) [inline src/redux/reducers/index.js]: case  SWITCH_THEME:
return state ===CUSTOM_THEMES.nightMode
                        ? CUSTOM_THEMES.defaultMode
                        : CUSTOM_THEMES.nightMode

File: src/redux/theme/index.js
Comment (andrzej-onja) [inline src/redux/theme/index.js]: try if it can be removed

File: src/redux/theme/index.js
Comment (andrzej-onja) [inline src/redux/theme/index.js:2]: add name or id prop : 'default' or 'dark'

File: src/screens/Home.js
Comment (andrzej-onja) [inline src/screens/Home.js]: set theme in initial state and remove this

File: src/components/Textarea/Textarea.js
Comment (andrzej-onja) [inline src/components/Textarea/Textarea.js]: this is weird `textColor === '#111827' ? 'rgba(17, 24, 39, 0.5)'`

File: src/components/Textarea/Textarea.js
Comment (andrzej-onja) [inline src/components/Textarea/Textarea.js]: color strings should be stored as consts as they are repeating so many times '#111827'

File: src/components/Textarea/Textarea.js
Comment (andrzej-onja) [inline src/components/Textarea/Textarea.js:12]: maybe just use default value `'#111827'`
`textColor='textColor,'`
and you can get rid of 
`color: textColor ? textColor : '#111827'`
and just use
`color: textColor`

**Review by andrzej-onja — APPROVED** [review]

**Review by rinonTen — APPROVED** [review]

**Review by mitia-anah — APPROVED** [review]

File: src/components/Textarea/Textarea.js
Comment (Fodilahy-mena) [reply src/components/Textarea/Textarea.js:26]: True!

File: src/components/Textarea/Textarea.js
Comment (Fodilahy-mena) [reply src/components/Textarea/Textarea.js:36]: This is the only way of approaching it. Maybe you do not understand the logic here. I am doing this because, for the components that do not have ` textColor` that passed to it, they use `"#111827"` color as a default. I what you mean about creating a variable but I don't think it's necessary.

File: src/redux/theme/index.js
Comment (Fodilahy-mena) [reply src/redux/theme/index.js:22]: I would not use variables to assign to an object prop value, especially in this case.

File: src/components/Textarea/Textarea.js
Comment (Fodilahy-mena) [reply src/components/Textarea/Textarea.js]: This is because we don't really have a specific color for placeholder in night mode.

**Pull request merged**
